nulib/dc-nextjs
# Support deep linking of user selected facets

## What does this do?

- Moves tracking/handling user selected facets out of the `SearchProvider` component.
- Current user facets are now read and updated by individual components, via `Next/Link`'s routing object.
- Tests updated to reflect a change in the `SearchProvider` component and supports mocking Next's Router package
- In the Search page component, we bypass the `useFetchApiData` hook and grab the information we want directly from the component.  There were a lot of timing errors relying upon the convenience hook (for some reason), and after trying a bunch of different approaches, this seemed to simplify things and produce consistent results.

<img width="917" alt="image" src="https://user-images.githubusercontent.com/3020266/189720638-57c62600-2c51-4001-8faf-e925b8cb8cf9.png">


## To Test
1. Go to the /search page
2. Try a combination of searching for terms, applying facets, removing facets, etc.
3. New search terms should wipe out previous facet selections.   
4. New facet selections should retain the search query, and narrow down the results
5. Try doing hard refreshes in the browser and notice we can save complex Search page filtering now
